internal/controller/scenario/scenario_adapter.go

```diff
@@ -103,48 +103,5 @@ func (a *Adapter) EnsureCreatedScenarioIsValid() (controller.OperationResult, er
 		a.logger.LogAuditEvent("IntegrationTestScenario marked as Valid", a.scenario, h.LogActionUpdate)
 	}
 
-	err := h.AddFinalizerToScenario(a.context, a.client, a.logger, a.scenario, h.IntegrationTestScenarioFinalizer)
-	if err != nil {
-		a.logger.Error(err, "Failed to add finalizer to IntegrationTestScenario")
-		return controller.RequeueWithError(err)
-	}
-
-	return controller.ContinueProcessing()
-}
-
-// EnsureDeletedScenarioResourcesAreCleanedUp is an operation that ensures that all resources related to the
-// deleted IntegrationTestScenario are cleaned up.
-func (a *Adapter) EnsureDeletedScenarioResourcesAreCleanedUp() (controller.OperationResult, error) {
-	if a.scenario.DeletionTimestamp == nil {
-		return controller.ContinueProcessing()
-	}
-
-	environmentsForScenario, err := a.loader.GetAllEnvironmentsForScenario(a.context, a.client, a.scenario)
-	if err != nil {
-		a.logger.Error(err, "Failed to find all Environments for IntegrationTestScenario")
-		return controller.RequeueWithError(err)
-	}
-	for _, testEnvironment := range *environmentsForScenario {
-		testEnvironment := testEnvironment
-		if h.IsEnvironmentEphemeral(&testEnvironment) {
-			dtc, err := a.loader.GetDeploymentTargetClaimForEnvironment(a.context, a.client, &testEnvironment)
-			if err != nil || dtc == nil {
-				a.logger.Error(err, "Failed to find deploymentTargetClaim defined in environment", "environment.Name", &testEnvironment.Name)
-				return controller.RequeueWithError(err)
-			}
-
-			err = h.CleanUpEphemeralEnvironments(a.context, a.client, &a.logger, &testEnvironment, dtc)
-			if err != nil {
-				a.logger.Error(err, "Failed to delete the Ephemeral Environment")
-				return controller.RequeueWithError(err)
-			}
-		}
-	}
-	// Remove the finalizer from the scenario since the cleanup has been handled
-	err = h.RemoveFinalizerFromScenario(a.context, a.client, a.logger, a.scenario, h.IntegrationTestScenarioFinalizer)
```

Comment:
But we have to make sure, that existing finalizers are removed, before we can drop code for removing finalizers

Reply:
Yup, I left a [note in the story](https://issues.redhat.com/browse/STONEINTG-815?focusedId=24707384&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-24707384) to do this manual step immediately after the changes are deployed.

Reply:
I don't like manual interventions, service itself should be done in a way to deal with it. If this is has multiple deployments as a real open source project, admins around the world won't say you anything nice about doing manual updates

Reply:
Ok, the alternative is to call `h.RemoveFinalizerFromScenario` as part of the [EnsureCreatedScenarioIsValid](https://github.com/konflux-ci/integration-service/blob/baf4328df8388e37980bfd52ebebd9c6c72b13aa/internal/controller/scenario/scenario_adapter.go#L58) - probably simply replacing the current [AddFinalizerToScenario](https://github.com/konflux-ci/integration-service/blob/baf4328df8388e37980bfd52ebebd9c6c72b13aa/internal/controller/scenario/scenario_adapter.go#L106-L110) call.
Since the function does a no-op if the finalizer is not there, I think that would work well to ensure backwards-compatibility.